### PR TITLE
Feat: update CLI plan command's create environment message

### PR DESCRIPTION
**Pull request by treysp**
The `sqlmesh plan` command's output includes a message about creating environments. 

The message can be confusing, so this PR updates it to be consistent and clear.

For different `plan` scenarios, the old and new messages are:

**Create prod env**
- Old 
	New environment `prod` will be created from `prod`
	Summary of differences against `prod`:
- New 
	`prod` environment will be initialized

**Create dev env without first creating prod**
- Old 
	New environment `dev` will be created from `prod`
	Summary of differences against `dev`:
- New 
	`dev` environment will be initialized

**New env with changes**
- Old
	New environment `dev` will be created from `prod`
	Summary of differences against `dev`:
- New
        New environment `dev` will be created from `prod`
        Differences from the `prod` environment:

**Existing env with changes**
- Old
        Summary of differences against `dev`:
- New
        Differences from the `dev` environment:

**Plan with no changes (existing environment)**
- Old
	No differences when compared to `prod`
- New
	No changes to plan: project files match the `prod` environment

**Plan with no changes (new environment)**
- Old
        Error: No changes were detected. Make a change or run with --include-unmodified to create a new environment without changes.
- New
        Error: Creating a new environment requires a change, but project files match the `prod` environment. Make a change or use the --include-unmodified flag to create a new environment without changes.

**Plan with --create-from environment that doesn't exist**
- Old
        New environment `dev` will be created from `doesnotexist`
        Summary of differences against `dev`:
- New
        WARNING - The environment name 'doesnotexist' was passed to the `plan` command's `--create-from` argument, but 'doesnotexist' does not exist. Initializing new environment 'dev' from scratch.
        `dev` environment will be initialized

### CODE REVIEW

File: sqlmesh/core/console.py [hunk 1]
```diff
@@ -613,22 +614,35 @@ def show_model_difference_summary(
             no_diff: Hide the actual SQL differences.
         """
         if context_diff.is_new_environment:
-            self._print(
-                Tree(
-                    f"[bold]New environment `{context_diff.environment}` will be created from `{context_diff.create_from}`"
-                )
+            msg = (
+                f"Initializing `{c.PROD}` environment"
```
Comment (izeigerman) [inline sqlmesh/core/console.py:618]: I think this message is misleading. We're just showing the plan summary here and so nothing gets "initialized" or "created" at this point. This is why it used to say "will be".

Comment (sungchun12) [reply sqlmesh/core/console.py:618]: so then you suggest? "`prod` environment will be created"

File: tests/cli/test_cli.py [hunk 2]
```diff
@@ -211,7 +215,6 @@ def test_plan_restate_model(runner, tmp_path):
     )
     assert result.exit_code == 0
     assert_duckdb_test(result)
-    assert "No differences when compared to `prod`" in result.output
```
Comment (izeigerman) [inline tests/cli/test_cli.py:214]: Should we check for `No changes to plan:`?

File: tests/cli/test_cli.py [hunk 3]
```diff
@@ -409,7 +454,6 @@ def test_plan_nonbreaking(runner, tmp_path):
         cli, ["--log-file-dir", tmp_path, "--paths", tmp_path, "plan"], input="y\n"
     )
     assert result.exit_code == 0
-    assert "Summary of differences against `prod`" in result.output
```
Comment (izeigerman) [inline tests/cli/test_cli.py:412]: Should we check for the new string here?

File: docs/examples/incremental_time_full_walkthrough.md [hunk 4]
```diff
@@ -292,7 +292,10 @@ Keep pressing enter on the date prompts, as we want to backfill all of history s
 ======================================================================
 Successfully Ran 2 tests against duckdb
 ----------------------------------------------------------------------
-Summary of differences against `dev`:
+New environment `dev` will be created from `prod`
```
[the remark this answers is not in the document]
Comment (izeigerman) [reply docs/examples/incremental_time_full_walkthrough.md:295]: `based on` may call for speculation

File: docs/concepts/models/seed_models.md [hunk 5]
```diff
@@ -129,7 +131,9 @@ $ sqlmesh plan
 ======================================================================
 Successfully Ran 0 tests against duckdb
 ----------------------------------------------------------------------
-Summary of differences against `prod`:
+Differences from the `prod` environment:
```
[the remark this answers is not in the document]
Comment (sungchun12) [reply docs/concepts/models/seed_models.md:134]: I'm fine either way on this. This is stylistic feedback at this point.

File: docs/examples/incremental_time_full_walkthrough.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sungchun12) [reply docs/examples/incremental_time_full_walkthrough.md:295]: The phrasing as is communicates clearly to me.

[the remark this answers is not in the document]
Comment (georgesittas) [reply docs/examples/incremental_time_full_walkthrough.md:295]: Sounds good, feel free to ignore.

File: docs/concepts/models/seed_models.md [hunk 6]
```diff
@@ -94,13 +94,15 @@ Christmas,2023-12-25
...
 When we run the `sqlmesh plan` command, the new seed model is automatically detected:
-```bash hl_lines="6-7"
+```bash hl_lines="8-9"
 $ sqlmesh plan
 ======================================================================
 Successfully Ran 0 tests against duckdb
 ----------------------------------------------------------------------
-Summary of differences against `prod`:
-└── Added Models:
+`prod` environment will be initialized
```
[the remark this answers is not in the document]
Comment (treysp) [reply docs/concepts/models/seed_models.md:102]: Good catch

**Review by izeigerman — APPROVED** [review]

**Pull request merged**